adi/pqmon.py

```diff
@@ -0,0 +1,547 @@
+# Copyright (C) 2022-2025 Analog Devices, Inc.
+#
+# SPDX short identifier: ADIBSD
+
+from adi.attribute import attribute
+from adi.context_manager import context_manager
+from adi.rx_tx import rx
+
+
+class pqmon(rx, context_manager, attribute):
+    """ The AD-PQMON-SL is a power quality monitoring solution based on the ADE9430,\n
+      which is a high performance, polyphase energy and Class S power quality monitoring \n
+      and energy metering device.
+    """
+
+    _device_name = "AD-PQMON-SL"
+
+    def __repr__(self):
+        retstr = f"""
+pqmon(uri="{self.uri}, device_name={self._device_name})"
+
+{self.__doc__}
+"""
+
+        return retstr
+
+    def __init__(self, uri="", device_name=""):
+
+        context_manager.__init__(self, uri)
+
+        found = False
+        for device in self._ctx.devices:
+            if device.name in "pqm":
+                self._ctrl = device
+                self._rxadc = device
+                self.rx_output_type = "SI"
+                found = True
+                break
+
+        if not found:
+            raise Exception("Could not find PQM de")
+
+        self.current_a = self.current_channel_phase(self._ctrl, "current0")
+        self.current_b = self.current_channel_phase(self._ctrl, "current1")
+        self.current_c = self.current_channel_phase(self._ctrl, "current2")
+        self.current_n = self.current_channel(self._ctrl, "current3")
+        self.voltage_a = self.voltage_channel(self._ctrl, "voltage0")
+        self.voltage_b = self.voltage_channel(self._ctrl, "voltage1")
+        self.voltage_c = self.voltage_channel(self._ctrl, "voltage2")
+
+        self.swell_events = []
+        self.dip_events = []
+        self.rvc_events = []
+        self.intrpt_events = []
+        self.config = self.config(self._ctrl, "config")
+
+        self._rx_channel_names = [
+            "voltage0",
+            "current0",
+            "voltage1",
+            "current1",
+            "voltage2",
+            "current2",
+            "current3",
+        ]
+        rx.__init__(self, 256)
```

Comment:
remove second argument since you are setting it on the next line. Next line is more readable